File: lib/models/transaction.dart
class Transaction {
  final String name;
  final String symbol;
  final int quantity;
  final double price;
  final DateTime timeStamp;
  final bool isBuy;

  Transaction({
    required this.name,
    required this.symbol,
    required this.quantity,
    required this.price,
    required this.timeStamp,
    required this.isBuy,
  });

  Map<String, dynamic> toMap() {
    return {
      'name': name,
      'symbol': symbol,
      'quantity': quantity,
      'price': price,
      'timeStamp': timeStamp,
    };
  }

  factory Transaction.fromMap(Map<String, dynamic> map) {
    return Transaction(
      name: map['name'] ?? '',
      symbol: map['symbol'] ?? '',
      quantity: map['quantity'] ?? 0,
      price: map['price'].toDouble() ?? 0.0,
      timeStamp: DateTime.parse(map['timeStamp']).toLocal(),
      isBuy: map['isBuy'] ?? false,
    );
  }
}

File: server/models/transaction.ts
import * as mongoose from "mongoose";

// Structure of Transaction Document
const transactionSchema = new mongoose.Schema({
  userId: { required: true, type: String, trim: true },
  symbol: { required: true, type: String, trim: true },
  name: { required: true, type: String, trim: true },
  quantity: { required: true, type: Number },
  price: { required: true, type: Number },
  timeStamp: { required: true, type: Date },
  isBuy: { required: true, type: Boolean },
});

// Convert schema into a model
const Transaction = mongoose.model("Transaction", transactionSchema);

export default Transaction;
